### PR TITLE
[unit-tests] Added more unit tests for OcTextInput component

### PR DESCRIPTION
**Pull request by kiranparajuli589**
## Description
- added unit tests for `OcTextInput` component

## Related Issue

- Fixes <issue_link>

## How Has This Been Tested?

- 💇🏼‍♂️ 
- 🤖 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation added/updated

## Open tasks:

- [ ] ...

### CODE REVIEW

File: src/components/OcTextInput.spec.js [hunk 1]
```diff
@@ -6,45 +6,190 @@ const defaultProps = {
...
+  })
+
+  describe("input classes", () => {
+    it("should be empty if stop class propagation is enabled", () => {
+      const wrapper = getShallowWrapper({ stopClassPropagation: true })
+      expect(wrapper.find("input").attributes().class).toBe("")
+    })
+  })
+
+  describe("when description message is provided", () => {
```
Comment (individual-it) [inline src/components/OcTextInput.spec.js:55]:
```suggestion
  describe("when a description message is provided", () => {
```

File: src/components/OcTextInput.spec.js [hunk 2]
```diff
@@ -6,45 +6,190 @@ const defaultProps = {
...
+    const wrapper = getShallowWrapper({ descriptionMessage: "You should pass." })
+    it("should add description class to input message", () => {
```
Comment (individual-it) [inline src/components/OcTextInput.spec.js:57]:
```suggestion
    it("should add the description class to the input message", () => {
```

File: src/components/OcTextInput.spec.js [hunk 3]
```diff
@@ -6,45 +6,190 @@ const defaultProps = {
...
+      expect(wrapper.find(selectors.textInputMessage).attributes().class).toContain(
+        "oc-text-input-description"
+      )
+    })
+    it("should show the description message as input message text", () => {
```
Comment (individual-it) [inline src/components/OcTextInput.spec.js:62]:
```suggestion
    it("should show the description message as the input message text", () => {
```

File: src/components/OcTextInput.spec.js [hunk 4]
```diff
@@ -6,45 +6,190 @@ const defaultProps = {
...
+      expect(wrapper.find(selectors.textInputMessage).text()).toBe("You should pass.")
+    })
+  })
+
+  describe("when warning message is provided", () => {
```
Comment (individual-it) [inline src/components/OcTextInput.spec.js:67]:
```suggestion
  describe("when a warning message is provided", () => {
```

File: src/components/OcTextInput.spec.js [hunk 5]
```diff
@@ -6,45 +6,190 @@ const defaultProps = {
...
+    const wrapper = getShallowWrapper({ warningMessage: "You may pass." })
+    it("should add warning class to input", () => {
```
Comment (individual-it) [inline src/components/OcTextInput.spec.js:69]:
```suggestion
    it("should add the warning class to the input", () => {
```

File: src/components/OcTextInput.spec.js [hunk 6]
```diff
@@ -6,45 +6,190 @@ const defaultProps = {
...
+      expect(wrapper.find("input").attributes().class).toContain("oc-text-input-warning")
+    })
+    it("should add warning class to input message", () => {
```
Comment (individual-it) [inline src/components/OcTextInput.spec.js:72]:
```suggestion
    it("should add the warning class to the input message", () => {
```

File: src/components/OcTextInput.spec.js [hunk 7]
```diff
@@ -6,45 +6,190 @@ const defaultProps = {
...
+      expect(wrapper.find(selectors.textInputMessage).attributes().class).toContain(
+        "oc-text-input-warning"
+      )
+    })
+    it("should show the warning message as input message text", () => {
```
Comment (individual-it) [inline src/components/OcTextInput.spec.js:77]:
```suggestion
    it("should show the warning message as the input message text", () => {
```

File: src/components/OcTextInput.spec.js [hunk 8]
```diff
@@ -6,45 +6,190 @@ const defaultProps = {
...
+      expect(wrapper.find(selectors.textInputMessage).text()).toBe("You may pass.")
+    })
+  })
+
+  describe("when error message is provided", () => {
```
Comment (individual-it) [inline src/components/OcTextInput.spec.js:82]:
```suggestion
  describe("when an error message is provided", () => {
```

File: src/components/OcTextInput.spec.js [hunk 9]
```diff
@@ -6,45 +6,190 @@ const defaultProps = {
...
+    const wrapper = getShallowWrapper({ errorMessage: "You shall not pass." })
+    it("should add error class to input", () => {
+      expect(wrapper.find("input").attributes().class).toContain("oc-text-input-danger")
+    })
+    it("should add error class to input message", () => {
```
Comment (individual-it) [inline src/components/OcTextInput.spec.js:87]:
```suggestion
    it("should add the error class to the input message", () => {
```

File: src/components/OcTextInput.spec.js [hunk 10]
```diff
@@ -6,45 +6,190 @@ const defaultProps = {
...
+      expect(wrapper.find(selectors.textInputMessage).attributes().class).toContain(
+        "oc-text-input-danger"
+      )
+    })
+    it("should show the error message as input message text", () => {
+      expect(wrapper.find(selectors.textInputMessage).text()).toBe("You shall not pass.")
+    })
+    it("should set input aria-invalid as true", () => {
```
Comment (individual-it) [inline src/components/OcTextInput.spec.js:95]:
```suggestion
    it("should set the input aria-invalid attribute to true", () => {
```

File: src/components/OcTextInput.spec.js [hunk 11]
```diff
@@ -6,45 +6,190 @@ const defaultProps = {
...
+        descriptionMessage: "Your should pass.",
+      })
+      const messageEl = wrapper.find(selectors.textInputMessage)
+      expect(messageEl.attributes().class).toBe("oc-text-input-description oc-text-input-warning")
+      expect(messageEl.text()).toBe("You may pass.")
+    })
+  })
+
+  describe("type prop", () => {
+    it("should not allow types other than text, number, email and password", () => {
```
Comment (individual-it) [inline src/components/OcTextInput.spec.js:125]:
```suggestion
    it("should only allow text, number, email and password as types", () => {
```

File: src/components/OcTextInput.spec.js [hunk 12]
```diff
@@ -6,45 +6,190 @@ const defaultProps = {
...
+        throw new Error(`Provided value 'binary' for prop 'type' is valid.`)
+      } catch (e) {
+        /* eslint-disable-next-line jest/no-conditional-expect, jest/no-try-expect */
+        expect(e).toContain(
+          '[Vue warn]: Invalid prop: custom validator check failed for prop "type".'
+        )
+      }
+    })
+    it.each(["text", "number", "email", "password"])(
+      "should set provided type for the input",
```
Comment (individual-it) [inline src/components/OcTextInput.spec.js:137]:
```suggestion
      "should set the provided type for the input",
```

File: src/components/OcTextInput.spec.js [hunk 13]
```diff
@@ -6,45 +6,190 @@ const defaultProps = {
...
+      type => {
+        const wrapper = getShallowWrapper({ type: type })
+        expect(wrapper.find("input").attributes("type")).toBe(type)
+      }
+    )
+  })
+
+  describe("input events", () => {
+    it("should emit input event on typing", async () => {
```
Comment (individual-it) [inline src/components/OcTextInput.spec.js:146]:
```suggestion
    it("should emit an input event on typing", async () => {
```

File: src/components/OcTextInput.spec.js [hunk 14]
```diff
@@ -6,45 +6,190 @@ const defaultProps = {
...
+      const wrapper = getShallowWrapper()
+      expect(wrapper.emitted().input).toBeFalsy()
+      await wrapper.find("input").setValue("a")
+      expect(wrapper.emitted().input).toBeTruthy()
+      expect(wrapper.emitted().input[0][0]).toBe("a")
+    })
+  })
+
+  describe("clear button accessible label prop", () => {
```
Comment (individual-it) [inline src/components/OcTextInput.spec.js:155]:
```suggestion
  describe("clear-button accessible label prop", () => {
```

File: src/components/OcTextInput.spec.js [hunk 15]
```diff
@@ -6,45 +6,190 @@ const defaultProps = {
...
+    it("should set aria label of clear input button if provided", () => {
```
Comment (individual-it) [inline src/components/OcTextInput.spec.js:156]:
```suggestion
    it("should set the aria label if provided", () => {
```

**Review by individual-it — CHANGES_REQUESTED** [review]
some grammatical improvements

**Comment by kiranparajuli589** [comment]
@individual-it changes has been made. please review again ♻️ .

**Review by individual-it — APPROVED** [review]

**Pull request merged**